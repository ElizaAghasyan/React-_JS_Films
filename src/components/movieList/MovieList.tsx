import styles from './MovieList.module.scss';
import MovieCard from "../movieCard/MovieCard";

interface Item {
    id: number;
    title: string;
    overview: string;
    poster_path: string;
    vote_average: number;
    backdrop_path: string;
    vote_count: number;
    name: string;
}

type MovieListProps = {
    item: Item[];
    category: {}
}

const MovieList = (props: MovieListProps) => {
    return (
        <div className={styles.cards}>
            {
                props.item?.map((item ) => (
                    <MovieCard key={item.id} item={item} />
                ))
            }
        </div>
    );
}

export default MovieList;
