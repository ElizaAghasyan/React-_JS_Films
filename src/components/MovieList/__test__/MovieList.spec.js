import MovieList from "../MovieList";
import renderer from "react-test-renderer";

describe('MovieList Component', () => {
    test('MovieList snapshot test', () => {
        const component = renderer.create(<MovieList />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
