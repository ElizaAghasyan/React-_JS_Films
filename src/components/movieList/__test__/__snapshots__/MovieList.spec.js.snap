// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movieList Component movieList snapshot test 1`] = `
<div
  className="cards"
/>
`;
